environment/src/main/java/com/sequenceiq/environment/environment/validation/securitygroup/aws/AwsEnvironmentSecurityGroupValidator.java

```diff
@@ -76,19 +77,25 @@ private void checkSecurityGroupVpc(EnvironmentDto environmentDto, ValidationResu
         CloudSecurityGroups securityGroups = platformParameterService.getSecurityGroups(request);
 
         boolean securityGroupInVpc = false;
-        for (CloudSecurityGroup cloudSecurityGroup : securityGroups.getCloudSecurityGroupsResponses().get(region.getName())) {
-            Object vpcId = cloudSecurityGroup.getProperties().get("vpcId");
-            if (cloudSecurityGroup.getGroupId().equals(securityGroupId)) {
-                LOGGER.info("Security group {} was found on AWS side.", securityGroupId);
-                if (vpcId != null && vpcId.toString().equals(environmentDto.getNetwork().getAws().getVpcId())) {
-                    securityGroupInVpc = true;
-                    break;
+        for (String securityGroupId : getSecurityGroupIds(securityGroupIds)) {
+            securityGroupInVpc = false;
+            for (CloudSecurityGroup cloudSecurityGroup : securityGroups.getCloudSecurityGroupsResponses().get(region.getName())) {
+                Object vpcId = cloudSecurityGroup.getProperties().get("vpcId");
+                if (cloudSecurityGroup.getGroupId().equals(securityGroupId)) {
+                    LOGGER.info("Security group {} was found on AWS side.", securityGroupId);
+                    if (vpcId != null && vpcId.toString().equals(environmentDto.getNetwork().getAws().getVpcId())) {
```

Comment:
Nit: `securityGroups.getCloudSecurityGroupsResponses().get(region.getName())` and `environmentDto.getNetwork().getAws().getVpcId()` do not change, so it would make sense to extract them into local variables outside the `getSecurityGroupIds()` loop of line 80.